src/hotel-search/client.js

```diff
@@ -65,7 +65,21 @@ HotelSearchClient.prototype = {
     var isSearchEnd = response.done || this.poller.isLastPolling()
     this.merger.mergeResponse(response, isSearchEnd);  
     this.lastRatesCount = response.count;
-    this.responseSearch = response.search;
+    this.responseSearch = this.prepareResponseSearch(response);
+  },
+
+  prepareResponseSearch: function(response) {
+    var responseSearch = response.search;
```

Comment:
Would be better if we let the `response` stay immutable. Though this can be `optional` as it won't cause any issues for now.

Reply:
Unlucky we don't follow 'immutable' concept and one of the main purpose of the sdk is to change and enrich response to make it ready and easy to use later.